src/Authenticator/CookieAuthenticator.php

```diff
@@ -154,7 +156,18 @@ protected function _createPlainToken($identity): string
         $usernameField = $this->getConfig('fields.username');
         $passwordField = $this->getConfig('fields.password');
 
-        return $identity[$usernameField] . $identity[$passwordField];
+        $salt = '';
+
+        if ($this->getConfig('salt')) {
+            if (is_string($this->getConfig('salt'))) {
+                $salt = $this->getConfig('salt');
+            } elseif (class_exists(Security::class)) {
+                $salt = Security::getSalt();
+                // Quietly skip salt if the Security utility is not available
+            }
```

Comment:
Never skip / hide issues. Let PHP generate an error so that the dev has useful information to act upon.

Reply:
If salt is enabled, a custom salt is not set and Security class is not available throw an exception.

Reply:
That makes sense. We should change the same here then:

https://github.com/cakephp/authentication/blob/aafb6831bd3a9e9f3607ffb254db529d5f6cf998/src/Authenticator/HttpDigestAuthenticator.php#L63-L65

Reply:
Why does that need to change? DigestAuthentication also ensures the secret is set https://github.com/cakephp/authentication/blob/aafb6831bd3a9e9f3607ffb254db529d5f6cf998/src/Authenticator/HttpDigestAuthenticator.php#L77-L80

Reply:
I misunderstood. I've copied a similar check.